### PR TITLE
Add 'security' as a sustainability principle

### PR DESCRIPTION
**Pull request by GeorgLink**
This pull request is to propose a new principle around security.

_Rationale:_ OSS is the digital infrastructure and society cannot afford OSS to be vulnerable. Sustainable OSS needs to reduce the risk of vulnerabilities.

The wording is open to discussion and change.

### CODE REVIEW

**Comment by adamhjk** [comment]
Thanks for this, @GeorgLink! I don't have any question about the sentiment - clearly, if what we build is insecure software, that's bad for the community at large. Right now, none of the principles say anything about the software in question - how it is developed, architecture, or process. That's on purpose, because we don't know what kind of community you are creating.

Talk to me about this from the angle of Rawls' game, and the impact it has on sustainable communities (and the institutions that support them). What about security brings it to the level of a foundational principle of a sustainable community?

**Comment by adamhjk** [comment]
For example, does software that is insecure not already fail the core commitment? does insecure software ever solve our problems?

**Comment by GeorgLink** [comment]
Thanks @adamhjk. Do you have a reference for "the angle of Rawls' game"? I am not familiar with this.

**Comment by adamhjk** [comment]
No problem, @GeorgLink. I'm talking about John Rawls, a political philosopher. You can read about how I used his game (the "veil of ignorance") to come up with the principles in the short book I wrote: https://sfosc.org/sfosc-book/ 

My question will make more sense after that :)

**Comment by jdorfman** [comment]
@adamhjk FWIW here is some context for this PR

**Comment by GeorgLink** [comment]
I agree with the principle and best practice separation (Twitter conversation). I still need to read the book before I attempt to formulate security as a principle. - I do think there is a technical security principle, similar to how we have community security principle (best practice: code of conduct)

**Comment by Beanow** [comment]
Going to have a got at approaching this with Rawls in mind. (TL;DR at the bottom)

You have the freedom to do the security audits yourself, fix any security problems yourself, etc. Courtesy of the four fundamental freedoms. So equal liberties :heavy_check_mark: check.

If we were to consider more security patches not as a basic liberty but a secondary benefit you could argue it's the same thing as any improvement that makes the code Harder Better Faster Stronger. The moment this improvement is freely shared with a free license, it's equally available to everyone. The same holds true to when the bug is never found or fixed, it's equally available to none. So I would conclude pushing security best practices doesn't make a difference in fairness / justice. It's just that everyone stands to benefit, in the same way that receiving more donations would be awesome, but no more or less fair.

I think problems in terms of Rawls start to appear when you don't share equally. And security is likely to fall prey to this. I'm talking mainly about disclosure here. When you uncover a vulnerability, you could argue it puts people in a few groups, in order of most to least advantaged position.

1. Those who know about it. And by extension in the best position to fix or exploit.
2. Those who don't know, but aren't vulnerable.
3. Those who don't know and are vulnerable.

We have some inequality now, but still meets Rawls criteria. Because you had equal opportunity to learn about security, do your own audit and find this information yourself. And nobody has leveraged it yet to actually realize the theoretical advantage.

When you use this information for fixes or exploits or studying though, you increase the inequality by getting additional benefits (the finished exploit or patch or knowledge). And according to Rawls, they are to be to the greatest advantage of the least advantaged members of society. Therefore the right thing to do is fix, followed by just study and worst is to exploit.

But there's a nuance. He uses **greatest** advantage. So how can we increase the advantage of a security fix for "those who don't know and are vulnerable".

1. Make sure as many people as possible are no longer vulnerable.
1. Make sure as many people as possible know about it in as much depth as you do.

Because in a perfect world that would put everyone back into the top position.
In the real world, not everyone is going to update their software, nor keep up with security news. But you can make a best effort to ensure it happens under the basic liberties + equal opportunity.

If you are unable to make such a fix, how can you create the greatest advantage for the least advantaged?

1. Find someone who can create a fix and disclose the information, while minimizing the risk of it being used for exploits, or study only.
1. Make sure as many people as possible are no longer vulnerable.
1. Make sure as many people as possible know about it in as much depth as you do.

Why minimize the risks? It makes intuitive sense, sharing a 0-day on the dark web, even if you didn't get paid for it, doesn't seem like the right thing to do. But my Rawls rationale for this is:
What's worse than the inequality of one or more persons knowing about a vulnerability? The vulnerability being exploited. If you share it carelessly, you would move from bad to worse.

And we have best practices for this, responsible disclosure.
I'll leave it as a readers exercise to see you could fit all the best practices into this line of thinking:
- Fix as fast as you reasonably can.
- Treat the disclosure with high confidentiality and security.
- Being careful about the timing of public statements.
- Incentivizing you to disclose with various rewards.
- Go public with most of the information eventually.

**My conclusion**

I think Rawls theory doesn't compel you to try very hard at making the software secure. But it has great benefits, so it's a good idea to try anyway. Optional but worthwhile.

But I do think it says you:
- MUST try and get vulnerabilities fixed and publicly disclosed when discovered.
- SHOULD have a responsible disclosure process in place.
- SHOULD want the guarantees of the responsible disclosure to be as strong as possible to reduce risk and ensure timely fixes and public disclosure.

Note: that first one implies you have a continuous moral obligation to make the fix and public disclosure happen. And not just the person who found it, everyone. You should pressure, lobby, monitor, or whatever other means you have available to make this process the best it can be.

Can you get away with not setting up responsible disclosure?
Well theoretically, I'd say yes. If no vulnerability is ever found, the process wouldn't make a difference. You're actually paying an opportunity cost for setting it up. But even a mostly effective hastily put together process beats none. And the cost for that is reading a few pages of background info, copy pasting a standard text into your repo and setting up a disclosure address.
It's like making regular backups, you're probably making excuses why you aren't and even a crappy backup is better than none.

File: content/principles/_index.md [hunk 1]
```diff
@@ -33,8 +33,9 @@ four freedoms for all, under all circumstances:
...
 * We must have a strong code of conduct, with clear, fair enforcement mechanisms.
 * Any patents included in the software must be granted under the terms of the open source license.
-* All contributors must retain their copyright, unless the software is being managed by a foundation for the purposes of license enforcement
+* All contributors must retain their copyright, unless the software is being managed by a foundation for the purposes of license enforcement.
 * All contributors intending to have their work incorporated into a distribution must contribute their work under the same terms as the software license they received it under.
 * Any commercial activity around the software must further the sustainability of the community, and the potential for commercial benefit must be available to all.
-* The incentives in any commercial model must bend away from the creation of proprietary downstream software
+* The incentives in any commercial model must bend away from the creation of proprietary downstream software.
+* All software must be developed with security in mind to avoid introducing vulnerabilities across the open source ecosystem.
```
Comment (nothingismagick) [inline content/principles/_index.md]: "developed with security in mind"
=> this is wishy-washy. I understand what you are saying, but think that there is more to it, as with what @Beanow discussed.

**Comment by Beanow** [comment]
Let's move this to the issue #85 to discuss this further and move to PRs when it's cleared up.

**Pull request closed without merging**